lib/inspec/config.rb

```diff
@@ -23,6 +23,8 @@ class Config
       shell_command
     }.freeze
 
+    @@valid_fields = %w{version cli_options credentials compliance reporter}
```

Comment:
Is there a reason why you went with class variable instead of constant or class instance variable?

Reply:
The config was not instantiated before input plugins were being registered, thus not class instance variable.   The value changes during runtime, thus not constant.  At least that was my thought processes.  I made a specific setter for this value, named for its intention.  I am amicable.